impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java

```diff
@@ -75,8 +81,23 @@ public class TestcontainersKafkaCluster implements Startable, KafkaCluster {
     private final DockerImageName zookeeperImage;
     private final KafkaClusterConfig clusterConfig;
     private final Network network = Network.newNetwork();
+    private final String name;
     private final ZookeeperContainer zookeeper;
-    private final Collection<KafkaContainer> brokers;
+
+    /**
+     * Map of kafka <code>node.id</code> to {@link KafkaContainer}.
+     * Protected by lock of {@link TestcontainersKafkaCluster itself.}
+     */
+    private final Map<Integer, KafkaContainer> brokers = new TreeMap<>();
+
+    /**
+     * Tracks ports that are in-use by this cluster, by listener.
+     * The inner map is a mapping of kafka <code>node.id</code> to a closed server socket, with a port number
+     * previously defined from the ephemeral range.
+     * Protected by lock of {@link TestcontainersKafkaCluster itself.}
+     */
+    private final Map<Listener, SortedMap<Integer, ServerSocket>> ports = Map.of(Listener.EXTERNAL, new TreeMap<>(),
```

Comment:
With the various changes to the port pre-allocation, do we need to pass around `ServerSocket` or should port allocation return int?